src/log_surgeon/finite_automata/RegexNFA.hpp

```diff
@@ -168,10 +229,37 @@ class RegexNFA {
     auto get_root() -> NFAStateType* { return m_root; }
 
 private:
+    /**
+     * Helper method for breadth-first traversal of the NFA.
+     * Adds a state to the queue and visited set if it hasn't been visited before.
+     * @param dest_state
+     * @param visited_states
+     * @param state_queue
+     */
+    static auto add_to_queue_and_visited(
+            RegexNFAByteState const* dest_state,
+            std::queue<RegexNFAByteState const*>& state_queue,
+            std::unordered_set<RegexNFAByteState const*>& visited_states
+    ) -> void;
+
     std::vector<std::unique_ptr<NFAStateType>> m_states;
     NFAStateType* m_root;
 };
 
+template <RegexNFAStateType state_type>
+auto PositiveTaggedTransition<state_type>::serialize(
+        std::unordered_map<RegexNFAByteState const*, uint32_t> const& state_ids
+) const -> std::string {
+    return fmt::format("{}[{}]", state_ids.at(get_dest_state()), get_tag());
```

Comment:
```suggestion
    return fmt::format("{}[{}]", state_ids.at(m_dest_state), m_tag);
```
Don't need to call the getter inside the member function right?

Reply:
Yea, must have missed some when moving serialize into the classes.